extensions/amp-viewer-integration/0.1/amp-viewer-integration.js

```diff
@@ -14,5 +14,71 @@
  * limitations under the License.
  */
 
+import {Messaging} from './messaging.js';
+import {listen} from '../../../src/event-helper';
+import {viewerForDoc} from '../../../src/viewer';
+import {user} from '../../../src/log';
+
+
 export class AmpViewerIntegration {
+
+  /**
+   * @param {!Window} win
+   */
+  constructor(win) {
+    /** @const @private {!Window} win */
+    this.win_ = win;
+  }
+
+  /**
+   * @return {!Promise}
+   */
+  init() {
+    return new Promise(resolve => {
+      const viewer = viewerForDoc(this.win_.document);
+      this.getHandshakePromise_(viewer)
+      .then(viewerOrigin => {
+        const messaging = new Messaging(this.win_.parent, viewerOrigin,
+            (type, payload, awaitResponse) => {
+              return viewer.receiveMessage(
+                type, /** @type {!JSONType} */ (payload), awaitResponse);
+            });
+        viewer.setMessageDeliverer((type, payload, awaitResponse) => {
```

Comment:
Can't this just be `viewer.setMessageDeliverer(messaging.sendRequest.bind(sendRequest), viewerOrigin)`.